XMega RTC timer problem.

KHeintz opened the pull request:
I think this solve the RTC timer problem.
I use this changes in wiring.c
# define CLK_RTCSRC_RCOSC32_gc (0x06<<1)// Set the internal 32.768kHZ oscillator as clock source for RTC.
# ifdef USE_RTC

/\* Turn on internal 32kHz. */
OSC.CTRL |= OSC_RC32KEN_bm; //cal RC Oscillator

do {
    /\* Wait for the 32kHz oscillator to stabilize. */
} while ( ( OSC.STATUS & OSC_RC32KRDY_bm ) == 0);

/\* Set internal 32.768kHZ oscillator as clock source for RTC. */
CLK.RTCCTRL = CLK_RTCSRC_RCOSC32_gc | CLK_RTCEN_bm;//32.768kHZ RC RTC clock
# if defined(USE_RTC)

do {
    /\* Wait until RTC is not busy. */
} while (  RTC.STATUS & RTC_SYNCBUSY_bm );

/\* Configure RTC period to 1 millisecond. */
RTC.PER = 129;//1ms 
RTC.CNT = 0;
RTC.COMP = 129; 
RTC.CTRL = ( RTC.CTRL & ~RTC_PRESCALER_gm ) | RTC_PRESCALER_DIV1_gc;

newbie15 committed:
Added a note about non-standard row offsets in LiquidCrystal.cpp

In my case I had to change it to make my 16x4 LCD work properly

Fixes #1333

Commit:
delayMicroseconds(): added support for 1Mhz, 12Mhz and 24Mhz

1Mhz is a default clock speed on Atmega328, many users run it on the
internal 1Mhz clock to save battery power. Up until now
delayMicroseconds() function wasn't taking this frequencies into an
account.

Commit:
zero delay fix, ADC prescaler fix

avishorp committed:
Allow the serial monitor to stay opened during upload, disabling it

bobc committed:
Add watchdog routines for Due.

bobc committed:
Add Due watchdog functions; modified according to feedback supplied

championswimmer committed:
update Sketch menu, add Save hex option

  * Moving Upload options from "File" menu to "Sketch" menu as those
     are sketch actions more than file actions.

Signed-off-by: Arnav Gupta <championswimmer@gmail.com>

NicoHood committed:
RX and TX Serial buffer sizes can now be defined independently

Fixes #2597

Commit:
Make the export option string translatable

championswimmer committed:
saveHex: rename recipe.hex -> recipe.output

Signed-off-by: Arnav Gupta <championswimmer@gmail.com>

facchinm committed:
Revert "Temporary fix for pulseIn() regression."

This reverts commit 8ddc5198f6ef8b2816a61b7b0fcce99f927d11a0.
To be substituted by ASM generated code

facchinm committed:
pulseIn: add ASM implementation

this assembly code was generated by avr-gcc 4.8.3

facchinm committed:
pulseIn: modify function to use ASM implementation

facchinm committed:
pulseIn: add alternative implementation based on micros()

pulseInLong is suitable for long pulses in interrupt context

facchinm committed:
pulseInASM: rework comment style

facchinm committed:
pulseIn: be consistent with standard API

return 0 if timeout has been reached

facchinm committed:
Due watchdog: change flags to effectively reset board

cmaglie committed:
[gemma] Integration of Gemma board

cmaglie committed:
[gemma] Added driver for Windows

ricardojlrufino committed:
removing files from the old editor

ricardojlrufino committed:
install script for linux

ricardojlrufino committed:
arduino logo for linux icon

ricardojlrufino committed:
build and run scripts - add new dependency and linux desktop icon

ricardojlrufino committed:
simple code refactor on PdePreprocessor

ricardojlrufino committed:
add RSyntaxTexarea jar

ricardojlrufino committed:
updated tests for new Textarea

ricardojlrufino committed:
improves font rendering

ricardojlrufino committed:
Allow redirect logs to console

ricardojlrufino committed:
new editor based on RSyntaxTextArea

ricardojlrufino committed:
update classpath - add new dependency

ricardojlrufino committed:
make editor folding happy

ricardojlrufino committed:
remove not used interface

ricardojlrufino committed:
preferences for syntax highlighting theme

Commit:
Added lib_sync, utility script

ricardojlrufino committed:
Add 'Code Folding' preferences and fix Font-Size changes

Commit:
Renamed compiler.warning_flags pref to compiler.warning_level
Provided a default value for compiler.warning_flags
Releasing avr core 1.6.5

Commit:
Windows: Compiler wasn't supplying Windows command line with the correct double quotes

Commit:
Windows: added comment to DoubleQuotedArgumentsOnWindowsCommandLine to explain the need for such a char replacement

Commit:
New editor: addede missing license headers on new files

Commit:
Revert "make editor folding happy"

This reverts commit e0926139fb61f7edbc38dd5bf7aaff1644e07a8f.

Commit:
Revert "improves font rendering"

This reverts commit c2d9a1b020f82fa82ef0df04765adc40e87d145f.

Commit:
Removed color theme selection: we currently support only one theme

Commit:
New editor: removed 1px border around editor

damellis committed:
Add SPI transactions to TFT library.

squelched committed:
SD: allow multiple SD instances

The SDClass class makes a reference to "SD.card" instead of just "card". SD is a global instance of SDClass.
This prevents any other instance of SDClass from functioning correctly.
The fix also allows SDClass to be used with an SD card which is removed and replaced, whereas previously, using the global instance SD did not allow this due to the limitation of begin() which cannot be called more than once.

Commit:
Disabling uncertified warning, leaving vids/pids

facchinm committed:
Merge pull request #3042 from facchinm/test_PR3037

SD: allow multiple SD instances

Commit:
Restoring keyword.txt loading and parsing. Added token type identifiers and related style in theme.txt

Commit:
Fixed failing PdeKeywords test
Autoformat was not saving caret position any more. Fixed

Commit:
Autoformat was generating two undo action, the first one being an empty editor. Tested and fixed

Commit:
Added rsyntaxtextarea license file

Commit:
MacOSX: default font size set to 12

Commit:
Ensure hyperlinks are coloured using theme.txt colour
Enabling [CTRL,CMD]+click to open hyperlinks

Commit:
PdeKeywords must be read before consulting command line args

Commit:
Logging Verifying/Uploading when using CLI

Commit:
Bundled package and library index soon become obsolete. Download a fresh copy each time the IDE is built

ricardojlrufino committed:
hyperlinks enabled by default on comment

Commit:
MacOSX: double click on a .ino file works again. Fixes #2888

Commit:
Code folding was not visible when line numbers were disabled. Fixed

Commit:
Removed left over println

Commit:
Removed stale static variable

Commit:
File.listFiles() can return null

Commit:
Core/Lib manager confirmation dialogs should pop up onto that manager modal

Commit:
ContributionIndexer now loads and merges content of files named package_SOMETHING_index.json

Commit:
Introducing GPG signature verification when loading main package_index file

Commit:
Copying bundled signature file if missing from user's preferences folder

Commit:
Added missing license header to tests

Commit:
Bundled package and library index: downloading gzipped json files at build time

Commit:
Boards Manager: downloading additional files as specified in preference 'boardsmanager.additional.urls'

Commit:
Boards Manager: index was updated twice. Fixed

Commit:
Boards Manager: package index file with name starting with "test_package_" are preserved

Commit:
Bridge: default baudrate is specified with BRIDGE_BAUDRATE. Fixes #2904

mschlenker committed:
add section BUGS

ffissore committed:
Merge pull request #3062 from mschlenker/master

Add section BUGS to manual page describing headless operation in Linux

Commit:
Boards Manager: platforms listed in non-signed package index files goes under "Contributed" category

Commit:
Boards/Library Managers: package and library index files get downloaded at build time even when building a "light bundle"

Commit:
ContributionInstaller: downloadIndex -> download

Commit:
Boards Manager: additional URLs may be specified from File > Preferences

Commit:
Lots of unclosed input and output streams now properly closed. They were preventing Boards Manager from working on Windows

Commit:
Examples: adding description, layout and schematics

ffissore committed:
Merge pull request #3055 from ffissore/other-packages

Allow listing other boards in boards manager

Commit:
Update revision.txt

Commit:
Mitigates #3074 by avoiding NPE. Making the IDE work with system toolchains remains to be fixed

Commit:
Deleting json files if they are some how corrupted. Fixes #3015

Commit:
GPGSignatureVerification: better error handling when reading signature

Commit:
Disabling uncertified boards warning even in java code, for good measure

Commit:
Preparing the ground for rewriting Preferences GUI code (and hopefully fixing the tiny-pref-window bug on macosx)

Commit:
Predicates: extracting anonymous inner classes for better reuse

Commit:
When IDE is started and built in core is newer than installed one and this check hasn't been notified before, show "please update" modal

Commit:
Restored Preferences static delegate methods and marking them as @Deprecated

Commit:
Added missing license headers to files added with the last commits

PaulStoffregen committed:
Add build symbols for compile time and sketch path

Commit:
MacOSX: fixed error when clicking on https urls. Fixes #3083

facchinm committed:
update revisions.txt

facchinm committed:
Merge pull request #3089 from facchinm/upd_rev

update revisions.txt

Commit:
Releasing 1.6.4

ffissore committed:
Merge pull request #3017 from cmaglie/programmer-pref-override

programmer.txt definitions must have priority over platforms.txt tools definitions

cmaglie committed:
Updated revision log

ffissore committed:
Merge pull request #3091 from cmaglie/gemma

Support for Arduino Gemma

Commit:
AVR core version bump

Commit:
Starting 1.6.5

Commit:
Export sketch: refactored to reuse existing code

Commit:
Ensuring sketch menu is properly rendered when editor windows gains/loses focus

Commit:
Fixing longstanding bug that made File and Tools menu items overlap "sometimes"

Commit:
Update revisions.txt

Commit:
Update revisions.txt

Commit:
Added missing licenses/copyrights

agdl committed:
Fixes  #2969:

This fixes the Wire examples that uses I2C reserved address (from 0 to 7) substituting them with 8 that is the first one available and that can be used.

I also modified the wire reference
http://www.arduino.cc/en/reference/wire
according to this fact.

Commit:
Code folding is disabled by default. Fixes #3105

Commit:
Added .editorconfig files http://editorconfig.org/

Commit:
Windows: this launch4j conf SHOULD force using only the bundled JVM

Commit:
Windows: launch4j urls changed to generic host downloads.sourceforge.net instead of switch.dl.sourceforge.net

facchinm committed:
Merge pull request #3111 from facchinm/test-watchdog-sam-2

Add watchdog routines for Due

Commit:
Added BaseNoGui.VERSION_NAME_LONG: it includes "Hourly Build DATE" in case of a hourly build

Commit:
Disabled port type fake menu entry. Fixes #3113

Commit:
Restoring left empty space on the editor

Commit:
Adding keywords. Closes #2213

Chris--A committed:
IDE: prevent multiple bad name notifications.

This patch will prevent the IDE from informing the user about a bad
sketch folder name multiple times.

Now the user is informed once at start up only, or once when it is
detected during a rescan of the folders.

Commit:
Update revisions.txt

Commit:
GNU/Linux: removing jayatana from JAVA_TOOL_OPTIONS, if present. Fixes #3081

Commit:
Windows: JVM is packaged with "build" (was previously packed later, with "dist")

Commit:
MacOSX: fixed some failing tests

Commit:
Avoid referring directly to RSyntaxTextArea, at least in the names

Commit:
New editor: restored CTRL+K (open sketch folder). Fixes #3103

Commit:
Using FileUtils.copyFile instead of dealing with raw streams

Commit:
Better naming and use of File constructor (avoids hardcoding system dependent path separator)

Commit:
New modal dialog for adding additional boards manager url: with a wide text area, inputing is much easier

Commit:
Update revisions.txt

Commit:
New editor: mark occurrences enable when "editor.advanced" (hidden pref) is true. Fixes #3102

Commit:
New editor: ALT+ UP/DOWN move current line only if "editor.advanced" (hidden pref) is true. Fixes #3101

Commit:
New editor: ALT+ BACKSPACE deletes next word (OSX only). See #3098

PaulStoffregen committed:
Fix Uncategorized warning message

Commit:
ArduinoIDE is in the default package. Removed

Commit:
New editor on MacOSX: CMD+BACKSPACE deletes current line until cursor position, ALT+BACKSPACE deletes previous word. See #3098

Commit:
New editor on MacOSX: CMD+UP/DOWN moves cursor to start or end of sketch. See #3098

ffissore committed:
Merge pull request #3130 from PaulStoffregen/uncategorized-warning

Fix Uncategorized warning message

ffissore committed:
Merge pull request #3109 from agdl/fix_i2cexamples

Fix for #2969 (I2C reserved addresses)

Commit:
New editor on MacOSX: restored CMD+E for finding selected text

ish6614 committed:
Updated cursor.ino

Previous commit broke the actual functionality. The code was confusing and worked until the previous commit.
The changes make it logically and functionally correct.

Commit:
Restored current line/current selected lines display on lower left of the IDE. Fixes #3134

cmaglie committed:
Removed buggy redundant check in FileUtils.deleteIfExists()

The documentation for File.delete() says that the method return true
if the the file is successfully deleted, otherwise false is returned.
An exception is thrown only when the file is not accessible (for
permission problem).

Removing the extra check solves another problem, for example in a
folder with the following situation:

    linkToFileA -> FileA
    FileA

if we remove FileA, we remain with a broken link that can't be removed
using FileUtils.deleteIfExists() because calling File.exists() on a
broken link returns *false*. This commit solve this problem.

cmaglie committed:
Removed redundant call to File.deleteIfExists()

file is already checked for being not null, no need to check again.

Chris--A committed:
Remove spawn from exec command

cmaglie committed:
Merge pull request #3135 from ish6614/patch-2

Updated cursor.ino

Commit:
New Preferences window: renders fine on every OS and it's easier to adapt using NetBeans as visual editor. Fixes #3140

Commit:
Old Preferences class remains for backwards compatibility as a delegate for PreferencesData

Commit:
New editor on MacOSX: since CMD+J is known as "jump to selection" and the editor has no such feature, CMD+J is disabled on mac. See #3098

ffissore committed:
Merge pull request #3141 from cmaglie/fix-recursive-delete

Fix for FileUtils.recursiveDelete() when dealing with symbolic links

mathertel committed:
Optimizations: remove multiple calls to the status() function.

Chris--A committed:
Updated keywords.txt

This commit fixes some highlighting issues due to multiple keyword
entries.

The missing C++ keywords have been added.

A set of standard fixed sized integers has been added also: uint8_t,
int32_t,...

Commit:
RSyntaxTextArea: using a modified version, tracked at https://github.com/arduino/RSyntaxTextArea. Fixes #3099

ffissore committed:
Merge pull request #3180 from Chris--A/keywords_update

Updated keywords.txt

ffissore committed:
Merge pull request #3167 from Chris--A/no-spawn

Build: Remove spawn from exec command (Windows)

Commit:
Don't export sketch if the underlying core does not support it. Fixes #3171

Commit:
Preferences: added tooltip about additional boards manager urls. Fixes #3092

Commit:
New editor on MacOSX: CMD+E uses selection to find next entry. Fixes #3147

Commit:
Libraries bundled with cores will take the type of that core, instead of relying on an undocumented 'types' property. Fixes #2875

Commit:
Rewritten library menu generation; libraries are now grouped by their type. Fixes #2879

Commit:
LibraryManagerUI: library types order fixed

Commit:
Libraries: fixed incorrect handling of null types

Commit:
Windows: addind -Djna.nosys=true to launcher options. Closes #2301

Commit:
Fixed goto keyword color. Fixes #3182

Commit:
Cut off board/port name at 50 instead of 16. Fixes #3104

Chris--A committed:
Added support for .hh and .hpp libraries

ffissore committed:
Merge pull request #3186 from Chris--A/allow_headers

Added support for .hh and .hpp libraries

cmaglie committed:
Added missing license on samd's pgmspace.h

Fixes #3145

Commit:
Update revisions.txt

ffissore committed:
Merge pull request #3185 from ffissore/board-port-cut-off

Cut off board/port name when greater than 50

Commit:
No need to use static variables

Chris--A committed:
Added String handling to SD class

Commit:
Windows: got rid on JNA, thus no more UnsatisfiedLinkError (at least when accessing native OS functions). Fixes #3190

Commit:
Several File.list() calls missed check for null return value. Fixed

Commit:
Due to website configuration changes, every url starting with http://arduino.cc has been changed to http://www.arduino.cc. Fixes #3191

Commit:
When exporting compiled binary of an example (readonly) sketch, users are forced to save it into their sketchbook. Fixes #3127

Commit:
Update revisions.txt

Chris--A committed:
Modified randomSeed, now uses unsigned long.﻿

Commit:
Restored background color when switching off "external editor" flag in preferences. Fixes #3197

agdl committed:
Fix for serialEvent example

In the example is stated that the function is run between one loop and the next, but actually the call to the function was missing.
The comment also state that the response can be delayed using a delay in the loop,so I think that the way it should be is so by only
adding a call to the function serialEvent as first operation in the loop. I so added this call.

ffissore committed:
Merge pull request #3194 from ffissore/windows-jna

Got rid on JNA

Commit:
Update revisions.txt

Commit:
Updated MenuScroller with latest from https://tips4java.wordpress.com/2009/02/01/menu-scroller/
Reviewed its getMaximumItems method (now getMaximumDrawableMenuItems) so that it returns a 1/3rd less the max number, avoiding a bad crash on linuxes. Fixes #3174

ffissore committed:
Merge pull request #3203 from agdl/SerialEventFix

Added missing function call to serialEvent example

Commit:
Test files should not be modified

Commit:
Update revisions.txt

Commit:
Moved DownloadableContribution and DownloadableContributionDownloader to parent package

Commit:
Removed old AutoFormat class, useless since we switched to AStyle

Commit:
Closing streams using IOUtils.closeQuietly
Fixed badly handled stream found in the meanwhile

Commit:
Faster library list downloading by downloading gzipped version

Commit:
Fixed bug with flushing

ffissore committed:
Merge pull request #2742 from Wackerbarth/Issue_2180

Allow the serial monitor to stay opened during upload, disabling it

Commit:
ZipFile and Socket are not Closable in java 1.6

Commit:
Serial monitor: added 230400 and 250000 baud rates. Fixes #2796 and #341

Commit:
Update revisions.txt

Chris--A committed:
Optimize printLn calls.﻿

Commit:
Removed lots of calls to BaseNoGui.getPlatform(): static is evil

cmaglie committed:
Indent fix.

cmaglie committed:
update revision log

Commit:
Fixed wrong storage of recently opened sketches

Commit:
MacOSX: set NSSupportsAutomaticGraphicsSwitching to true, so the OSX can switch back to the integrated GPU and save battery. Fixes #3210

Chris--A committed:
Updated printLn on SAM core

damellis committed:
Don't find on Cmd+E, just set find text (Mac OS X).

vicatcu committed:
Add missing case to detachinterrupt

Commit:
Revert "MacOSX: set NSSupportsAutomaticGraphicsSwitching to true, so the OSX can switch back to the integrated GPU and save battery. Fixes #3210"

This reverts commit 45b61ef02bd139180a745b642385789a8cd69ce2.

Commit:
Allowing editing .S files. Fixes #1616 and #3059

Commit:
Stop using [ and ] as indentation menu accelerators. Use TAB/SHIFT+TAB instead. CMD+BRAKETS are still available on MacOSX only
Fixes #3224 and #220
Also see https://github.com/processing/processing/issues/2199

Commit:
Removed Preferences.init: no one used it

Commit:
CollectStdOutExecutor and CollectStdOutStdErrExecutor were plain wrong, were losing data and were blocking compilation. Fixes #3124 and #3115

Commit:
Windows: additional metadata to .exe

cmaglie committed:
Merge branch 'fixdetachinterrupt' of https://github.com/vicatcu/Arduino

Commit:
Scrollable menus can be a little higher

ffissore committed:
Merge pull request #2652 from swordmaster2k/master

Fix for #2402

cmaglie committed:
Merge branch 'master' of github.com:arduino/Arduino

cmaglie committed:
update revisions log

Commit:
MacOSX: when closing, no confirmation is displayed any more, unless at least one sketch is modified. In essence: now mac behaves just like the other OSs. Fixes #3093

ffissore committed:
Merge pull request #3227 from ffissore/editor-assembly-support

Allowing editing .S files

Commit:
Missing call to BaseNoGui.getPlatform().init() in tests

Commit:
Compiler: adding hooks (custom recipes) between the main phases. New hooks are:
hooks.sketch.prebuild, hooks.sketch.postbuild, hooks.libraries.prebuild,
hooks.libraries.postbuild, hooks.core.prebuild, hooks.core.postbuild,
hooks.linking.prelink, hooks.linking.postlink, hooks.objcopy.preobjcopy,
hooks.objcopy.postobjcopy, hooks.savehex.presavehex, hooks.savehex.postsavehex

Commit:
Avoid warning about SCCS folders in libraries, just ignore them. Fixes #3237

Commit:
Toogle comments even using the DIVIDE key on the keypad, so non US keyboard can use the shortcut. Fixes #431 and https://code.google.com/p/arduino/issues/detail?id=76

Commit:
MacOSX: millis arg was ignored

Commit:
Saving a couple of variables won't make the IDE any faster, but will make the dev slower

Lauszus committed:
Check if INT2 is actually defined before issuing warning in detachInterrupt

This was caused after: https://github.com/arduino/Arduino/pull/3221

facchinm committed:
Merge pull request #3207 from chromhelm/master

Fixed bug with HWSerial flushing

cmaglie committed:
Merge pull request #3244 from Lauszus/INT2

Check if INT2 is actually defined before issuing warning in detachInterrupt

facchinm committed:
Merge pull request #3211 from Chris--A/println-optimize

Println optimize

facchinm committed:
Merge pull request #3193 from Chris--A/SD_Strings

Added String handling to SD class

facchinm committed:
Merge pull request #3202 from Chris--A/randomSeed

Modified randomSeed, now uses unsigned long.﻿

Commit:
Do not cache PrinterJob instances. Cache is evil

Commit:
Removed a handful of static methods delegates

Commit:
SHIFT-save (upper toolbar) triggers "save as". Fixes #2220

ffissore committed:
Merge pull request #3236 from ffissore/build-hooks

Additional build hooks

ffissore committed:
Merge pull request #3229 from ffissore/indent-with-tab-no-brackets

Stop using [ and ] as indentation menu accelerators

agdl committed:
Merge pull request #3036 from damellis/tft-spi-transactions

Add SPI transactions to TFT library.

cmaglie committed:
update revision log

Commit:
Boards Manager: contributions MAY miss tools, but we missed to consider this case. Fixes #3248

Commit:
Changing order of boards in boards menu: built-in first, boards manager second,
sketchbook hardware folder third. Each group if sorted by alpha, case
insensitive. Fixes #2650

facchinm committed:
pulseIn: add ASM implementation for Due

facchinm committed:
pulseIn: add alternative implementation based on micros()

Commit:
ContributedPlatform.toString() is used as item in dropdown, so it should print only its version

Commit:
Command line --get-pref may be used without name of the target pref.
If pref name is missing, all prefs are printed on stdout.
Should fix/mitigate #2982

Commit:
Linux: arduino startup script uses #!/usr/bin/env bash. Fixes #3003

facchinm committed:
Merge branch 'master' into ide-1.5.x

facchinm committed:
Merge pull request #2495 from facchinm/test-pulseIn-pullreq2

rework pulseIn function to solve issue #2409

facchinm committed:
update revisions.txt

karlbackstrom committed:
Merge remote-tracking branch 'upstream/ide-1.5.x' into xmegaduino

karlbackstrom committed:
Add support for creating package and use new Board Manager in official Arduino.

karlbackstrom commented:
Hmm, the pull request seems to be against master branch, could you send it for the xmegaduino branch instead?

zdjurisic committed:
Update platform.txt

Updated compiler.path and patterns to what is appropriate for compiler supplied with Arduino IDE 1.8.2. Compilation would update those on it's own during compilation anyway; this update just removes warning messages.

zdjurisic committed:
Update wiring.c

in iox___.h, for some XMegas, analog reference (internal Vcc/1.6) is referred to as ADC_REFSEL_VCC_gc and for some as ADC_REFSEL_INTVCC_gc This edit covers all XMegas supplied with Arduino IDE 1.8.2 (avrdude 6.3.0, avr-gcc 4.9.2)

karlbackstrom committed:
Merge pull request #46 from zdjurisic/patch-1

Update platform.txt

karlbackstrom committed:
Merge pull request #47 from zdjurisic/patch-2

Update wiring.c
